### ISSUE TITLE
Exception for space-before-keywords?

### ISSUE THREAD

**Issue by olalonde** [airbnb/javascript#713]
Writing a React/redux app, I am occasionally declaring anonymous classes inside function calls (which was fine before updating the airbnb config recently):

```javascript
export default connect((s) => s)(class extends Component {
  // React component class definition
});
```

The `space-before-keywords` rule makes the code above look a bit ugly:

```javascript
export default connect((s) => s)( class extends Component {
  // React component class definition
});
```

Was wondering if that was an unintended effect of the `space-before-keywords` or if everything is working as expected.

**Comment by tleunen** [comment]
I think the styleguide should specify to avoid anonymous classes. Especially in case of React where the class name is used as a "display name".

**Comment by ljharb** [comment]
I agree one should avoid anonymous classes - but also, aggressive inlining. I'd recommend defining the class first, and then exporting the connected value in a second line.

Specifically, I think this is a bug with the `space-before-keywords` linter rule, that perhaps should be filed to eslint directly.
